{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ba2faf4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import regex\n",
    "import pandas as pd\n",
    "from function import revers,replace_ambiguity_codes\n",
    "\n",
    "def revers(seq):\n",
    "    codes={\"A\":\"A\", \"C\":\"C\", \"G\":\"G\", \"T\":\"T\", \n",
    "           \"R\":\"[AG]\", \"S\":\"[GC]\", \"B\":\"[CGT]\", \"Y\":\"[CT]\", \n",
    "           \"W\":\"[AT]\", \"D\":\"[AGT]\", \"K\":\"[GT]\", \"N\":\"[ACGT]\", \n",
    "           \"H\":\"[ACT]\", \"M\":\"[AC]\", \"V\":\"[ACG]\", \"X\": \"[ACGT]\"}\n",
    "    \n",
    "    d={'A':'T','G':'C','C':'G','T':'A',\n",
    "      'R':'Y','S':'S','B':'V','Y':'R',\n",
    "       'W':'W','D':'H','K':'M','N':'N',\n",
    "       'H':'D','M':'K','V':'B','X':'X'\n",
    "      }\n",
    "    new=''\n",
    "    for i in seq[::-1]:\n",
    "        new+=d[i]\n",
    "    return new\n",
    "\n",
    "def replace_ambiguity_codes(sequence):\n",
    "    codes={\"A\":\"A\", \"C\":\"C\", \"G\":\"G\", \"T\":\"T\", \n",
    "           \"R\":\"[AG]\", \"S\":\"[GC]\",\"B\":\"[CGT]\", \"Y\":\"[CT]\", \"W\":\"[AT]\",\n",
    "           \"D\":\"[AGT]\", \"K\":\"[GT]\", \"N\":\"[ACGT]\", \"H\":\"[ACT]\", \"M\":\"[AC]\", \"V\":\"[ACG]\", \"X\": \"[ACGT]\"}\n",
    "    regexlist=[]\n",
    "    for base in sequence:\n",
    "        if not base in codes:\n",
    "            print(\"Unidentified nucleotide code in primer:\", base)\n",
    "            sys.exit()\n",
    "        else:\n",
    "            regexlist.append(codes[base])\n",
    "\n",
    "    return ''.join(regexlist)\n",
    "\n",
    "def getposition(match,matches):\n",
    "    \n",
    "    a = re.finditer(match, matches)\n",
    "    for match in a:\n",
    "        start_pos = match.start()\n",
    "        end_pos = match.end()\n",
    "\n",
    "        # Get the matched text itself\n",
    "        matched_text = match.group()\n",
    "\n",
    "        # Print the match and its positions\n",
    "        aa=f\"Match: '{matched_text}' | Start: {start_pos} | End: {end_pos}\"\n",
    "        return aa\n",
    "def coverage(forward_list, reverse_list, d):\n",
    "    coverage ={}\n",
    "    for each_class in d:\n",
    "        c=0\n",
    "        for i in d[each_class]:\n",
    "            f=0\n",
    "            r=0\n",
    "            pcr_product=[]\n",
    "            a=[];b=[]\n",
    "            for forward_primer in forward_list: \n",
    "                f_b = re.findall(forward_primer,i)\n",
    "                if len(f_b) == 1:\n",
    "                    a.append(getposition(f_b[0],i))\n",
    "            for reverse_primer in reverse_list:\n",
    "                r_b = re.findall(revers(reverse_primer),i)\n",
    "                if len(r_b) == 1:\n",
    "                    b.append(getposition(r_b[0],i))\n",
    "            if len(a)>0 and len(b)>0:\n",
    "                c+=1\n",
    "        coverage[each_class]=format(c*100/len(d[each_class]),'.1f')+'%'\n",
    "    return coverage\n",
    "def coverage_e1(forward_list, reverse_list, d):\n",
    "    coverage ={}\n",
    "    for each_class in d:\n",
    "        c=0\n",
    "        for i in d[each_class]:\n",
    "            f=0\n",
    "            r=0\n",
    "            pcr_product=[]\n",
    "            a=[];b=[]\n",
    "            for forward_primer in forward_list: \n",
    "                f_b = regex.findall(\"(\"+forward_primer+\"){e<=1}\",i)\n",
    "                if len(f_b) == 1:\n",
    "                    a.append(getposition(f_b[0],i))\n",
    "            for reverse_primer in reverse_list:\n",
    "                r_b = regex.findall(\"(\"+revers(reverse_primer)+\"){e<=1}\",i)\n",
    "                if len(r_b) == 1:\n",
    "                    b.append(getposition(r_b[0],i))\n",
    "            if len(a)>0 and len(b)>0:\n",
    "                c+=1\n",
    "        coverage[each_class]=format(c*100/len(d[each_class]),'.1f')+'%'\n",
    "    return coverage\n",
    "\n",
    "def cov(forward_primer, reverse_primer, d):\n",
    "    coverage ={}\n",
    "    for each_class in d:\n",
    "        c=0\n",
    "        for i in d[each_class]:\n",
    "            f=0\n",
    "            r=0\n",
    "            pcr_product=[]\n",
    "            a=[];b=[]\n",
    "            f_b = re.findall(replace_ambiguity_codes(forward_primer),i)\n",
    "            if len(f_b) == 1:\n",
    "                a.append(getposition(f_b[0],i))\n",
    "            r_b = re.findall(replace_ambiguity_codes(revers(reverse_primer)),i)\n",
    "            if len(r_b) == 1:\n",
    "                b.append(getposition(r_b[0],i))\n",
    "            if len(a)>0 and len(b)>0:\n",
    "                c+=1\n",
    "        coverage[each_class]=format(c*100/len(d[each_class]),'.1f')+'%'\n",
    "    return coverage\n",
    "def cov_e1(forward_primer, reverse_primer, d):\n",
    "    coverage ={}\n",
    "    for each_class in d:\n",
    "        c=0\n",
    "        for i in d[each_class]:\n",
    "            f=0\n",
    "            r=0\n",
    "            pcr_product=[]\n",
    "            a=[];b=[]\n",
    "            if 2>1:\n",
    "                f_b = regex.findall(\"(\"+replace_ambiguity_codes(forward_primer)+\"){e<=1}\",i)\n",
    "                if len(f_b) == 1:\n",
    "                    a.append(getposition(f_b[0],i))\n",
    "            if 2>1:\n",
    "                r_b = regex.findall(\"(\"+replace_ambiguity_codes(revers(reverse_primer))+\"){e<=1}\",i)\n",
    "                if len(r_b) == 1:\n",
    "                    b.append(getposition(r_b[0],i))\n",
    "            if len(a)>0 and len(b)>0:\n",
    "                c+=1\n",
    "        coverage[each_class]=format(c*100/len(d[each_class]),'.1f')+'%'\n",
    "    return coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93566794",
   "metadata": {},
   "source": [
    "# SILVA r138.1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e944839",
   "metadata": {},
   "outputs": [],
   "source": [
    "# p: SIlVA phylum level dict\n",
    "# c: SILVA Class level dict\n",
    "\n",
    "p={};c={}\n",
    "top20=[\"Bacteroidota\",\n",
    "       \"Proteobacteria\",\n",
    "       \"Actinobacteriota\",\n",
    "       \"Chloroflexi\",\n",
    "       \"Verrucomicrobiota\",\n",
    "       \"Myxococcota\",\n",
    "       \"Firmicutes\",\n",
    "       \"Planctomycetota\",\n",
    "       \"Acidobacteriota\",\n",
    "       \"Campylobacterota\",\n",
    "       \"Nitrospirota\",\n",
    "       \"Patescibacteria\",\n",
    "       \"Desulfobacterota\",\n",
    "       \"Bdellovibrionota\",\n",
    "       \"Cyanobacteria\",\n",
    "       \"Spirochaetota\",\n",
    "       \"Fusobacteriota\",\n",
    "       \"Euryarchaeota\",\n",
    "       \"Latescibacterota\",\n",
    "       \"Armatimonadota\"]\n",
    "for i in top20:\n",
    "    p[i]=[]\n",
    "p['Others'] = []\n",
    "#https://www.arb-silva.de/no_cache/download/archive/release_138.1/Exports/\n",
    "with open(r'SILVA_138.1_SSURef_NR99_tax_silva.fasta') as f:\n",
    "    data=('\\n'+f.read()).split('\\n>')[1:]\n",
    "    for i in data:\n",
    "        if 'Patescibacteria' in i:\n",
    "            if i.split('\\n')[0].split(';')[2] not in c:\n",
    "                c[i.split('\\n')[0].split(';')[2]] = [i.split('\\n',1)[1].replace('U','T').replace('\\n','')]\n",
    "            else:\n",
    "                c[i.split('\\n')[0].split(';')[2]].append(i.split('\\n',1)[1].replace('U','T').replace('\\n',''))\n",
    "        if 'Eukaryota' not in i:\n",
    "            if i.split('\\n')[0].split(';')[1] not in top20:\n",
    "                p['Others'].append(i.split('\\n',1)[1].replace('U','T').replace('\\n',''))\n",
    "            else:\n",
    "                p[i.split('\\n')[0].split(';')[1]].append(i.split('\\n',1)[1].replace('U','T').replace('\\n',''))\n",
    "c2 = {}\n",
    "c2['Others'] = []\n",
    "common_class=['Parcubacteria', 'Gracilibacteria', 'Microgenomatia', 'Berkelbacteria', 'ABY1', 'Saccharimonadia', 'Dojkabacteria', 'WWE3']\n",
    "for i in c:\n",
    "    if i in common_class:\n",
    "        c2[i] = c[i]\n",
    "    else:\n",
    "        for j in c[i]:\n",
    "            c2['Others'].append(j)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dafc838b",
   "metadata": {},
   "source": [
    "# MiDAS 4.8.1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af7d5412",
   "metadata": {},
   "outputs": [],
   "source": [
    "# midas : MIDAS phylum dict contain seqs\n",
    "# mc(2): MiDAS Class level dict contain seqs\n",
    "midas={}\n",
    "mc = {}\n",
    "top20=[\"Bacteroidota\",\n",
    "       \"Proteobacteria\",\n",
    "       \"Actinobacteriota\",\n",
    "       \"Chloroflexi\",\n",
    "       \"Verrucomicrobiota\",\n",
    "       \"Myxococcota\",\n",
    "       \"Firmicutes\",\n",
    "       \"Planctomycetota\",\n",
    "       \"Acidobacteriota\",\n",
    "       \"Campylobacterota\",\n",
    "       \"Nitrospirota\",\n",
    "       \"Patescibacteria\",\n",
    "       \"Desulfobacterota\",\n",
    "       \"Bdellovibrionota\",\n",
    "       \"Cyanobacteria\",\n",
    "       \"Spirochaetota\",\n",
    "       \"Fusobacteriota\",\n",
    "       \"Euryarchaeota\",\n",
    "       \"Latescibacterota\",\n",
    "       \"Armatimonadota\"]\n",
    "for i in top20:\n",
    "    midas[i]=[]\n",
    "midas['Others'] = []\n",
    "# can be accessed here: https://www.midasfieldguide.org/guide/downloads\n",
    "with open(r'SINTAX fa file MiDAS 4.8.1.fa') as f:\n",
    "    data=('\\n'+f.read().replace(\"tax=\",'')).split('\\n>')[1:]\n",
    "    for i in data:\n",
    "        if i.split(',')[1].split('\\n')[0].replace('p:','') in top20:\n",
    "            midas[i.split(',')[1].split('\\n')[0].replace('p:','')].append(i.split('\\n',1)[1].replace('\\n',''))\n",
    "        else:\n",
    "            midas['Others'].append(i.split('\\n',1)[1].replace('\\n',''))\n",
    "        if 'Patesci' in i:\n",
    "            if i.split(',')[2].replace('c:','') not in mc:\n",
    "                mc[i.split(',')[2].replace('c:','')] = [i.split('\\n',1)[1].replace('\\n','')]\n",
    "            else:\n",
    "                mc[i.split(',')[2].replace('c:','')].append(i.split('\\n',1)[1].replace('\\n',''))\n",
    "mc2= {}\n",
    "mc2['Others'] = []\n",
    "common_class=['Parcubacteria', 'Gracilibacteria', 'Microgenomatia', 'Berkelbacteria', 'ABY1', 'Saccharimonadia', 'Dojkabacteria', 'WWE3']\n",
    "\n",
    "for i in mc:\n",
    "    if i in common_class:\n",
    "        mc2[i] = mc[i]\n",
    "    else:\n",
    "        for j in mc[i]:\n",
    "            mc2['Others'].append(j)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a0abc11",
   "metadata": {},
   "source": [
    "# primer sequences used in analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93ecc28c",
   "metadata": {},
   "outputs": [],
   "source": [
    "V4_515Yp = '''GTGCCAGCAGCAGCGGTAA\n",
    "GTGCCAGCACCCGCGGTAA\n",
    "GTGCCAGCAGACGCGGTAA\n",
    "GTGCCAGCAGCCGCGGTAA\n",
    "GTGCCAGCAGCCGCGGTCA\n",
    "GTGCCAGCAGCCGCGGTGA\n",
    "GTGCCAGCAGCTGCGGTAA\n",
    "GTGCCAGCAGTCGCGGCAA\n",
    "GTGCCAGCAGTCGCGGTAA\n",
    "GTGCCAGCCGCCGCGGTAA\n",
    "GTGGCAGCCGCCGCGGTAA\n",
    "GTGTCAGCAGCCGCGGTAA\n",
    "GTGTCAGCCGCCGCGGTAA'''.split('\\n')\n",
    "V4_926Rp = '''CCGTCAATTCGTTTAAGTTT\n",
    "CCACCAGAACATTTAAGTTT\n",
    "CCATCAATTCATTTGAGTTT\n",
    "CCGCCAATTCATTTAAGTTT\n",
    "CCGCCAATTCATTTGAGTTT\n",
    "CCGCCAATTCCCTTAAGTCT\n",
    "CCGCCAATTCCCTTAAGTTT\n",
    "CCGCCAATTCCTATAAGTTT\n",
    "CCGCCAATTCCTTCAAGTTT\n",
    "CCGCCAATTCCTTTAAGTCG\n",
    "CCGCCAATTCCTTTAAGTCT\n",
    "CCGCCAATTCCTTTAAGTTA\n",
    "CCGCCAATTCCTTTAAGTTT\n",
    "CCGCCAATTCCTTTGAGTTT\n",
    "CCGCCAATTGGTTTAAGTTT\n",
    "CCGCCAATTTCTTTAAGTTT\n",
    "CCGCCAATTTCTTTGAGTTT\n",
    "CCGTCAACTCATTTGAGTTT\n",
    "CCGTCAATCCCTTTAAGTTT\n",
    "CCGTCAATGCCTTTAAGTTT\n",
    "CCGTCAATTACTTTAAGTTT\n",
    "CCGTCAATTACTTTGAGTTT\n",
    "CCGTCAATTCATTTAAGTTT\n",
    "CCGTCAATTCATTTGAGTTT\n",
    "CCGTCAATTCCCTTGAGTTT\n",
    "CCGTCAATTCCTTCAAGTTT\n",
    "CCGTCAATTCCTTTAAATTT\n",
    "CCGTCAATTCCTTTAAGTTT\n",
    "CCGTCAATTCCTTTGAATTT\n",
    "CCGTCAATTCCTTTGAGTTC\n",
    "CCGTCAATTCCTTTGAGTTT\n",
    "CCGTCAATTCCTTTGGGTTC\n",
    "CCGTCAATTCCTTTGGGTTT\n",
    "CCGTCAATTCTTTTGAGTTT\n",
    "CCGTCAATTTATTTGAGTTT\n",
    "CCGTCAATTTCTTTAAGTTT\n",
    "CCGTCAATTTCTTTGAGTTT\n",
    "CCGTCTATTCCTTTGAGTTT'''.split('\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "48c847fc",
   "metadata": {},
   "source": [
    "# Coverage calculate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "de691513",
   "metadata": {},
   "outputs": [],
   "source": [
    "# _0 : 0 mismatch\n",
    "# _1: 1 mismatch\n",
    "silva_class_0 = coverage(V4_515Yp,V4_926Rp,c2)\n",
    "silva_phylum_0 = coverage(V4_515Yp,V4_926Rp,p)\n",
    "silva_class_1 = coverage_e1(V4_515Yp,V4_926Rp,c2)\n",
    "silva_phylum_1 = coverage_e1(V4_515Yp,V4_926Rp,p)\n",
    "midas_class_0 = coverage(V4_515Yp,V4_926Rp,mc2)\n",
    "midas_phylum_0 = coverage(V4_515Yp,V4_926Rp,midas)\n",
    "midas_class_1 = coverage_e1(V4_515Yp,V4_926Rp,mc2)\n",
    "midas_phylum_1 = coverage_e1(V4_515Yp,V4_926Rp,midas)\n",
    "\n",
    "v13_p_0=cov('AGAGTTTGATCCTGGCTCAG','ATTACCGCGGCTGCTGG',p) #27F 534R \n",
    "v13_p_1=cov_e1('AGAGTTTGATCCTGGCTCAG','ATTACCGCGGCTGCTGG',p) #27F 534R \n",
    "v13_midas_0=cov('AGAGTTTGATCCTGGCTCAG','ATTACCGCGGCTGCTGG',midas) #27F 534R \n",
    "v13_midas_1=cov_e1('AGAGTTTGATCCTGGCTCAG','ATTACCGCGGCTGCTGG',midas) #27F 534R \n",
    "\n",
    "v4_p_0=cov('GTGYCAGCMGCCGCGGTAA','GGACTACNVGGGTWTCTAAT',p) #515F (Parada)–806R (Apprill)\n",
    "v4_p_1=cov_e1('GTGYCAGCMGCCGCGGTAA','GGACTACNVGGGTWTCTAAT',p) #515F (Parada)–806R (Apprill)\n",
    "v4_midas_0=cov('GTGYCAGCMGCCGCGGTAA','GGACTACNVGGGTWTCTAAT',midas) #515F (Parada)–806R (Apprill)\n",
    "v4_midas_1=cov_e1('GTGYCAGCMGCCGCGGTAA','GGACTACNVGGGTWTCTAAT',midas) #515F (Parada)–806R (Apprill)\n",
    "\n",
    "v34_p_0=cov('CCTACGGGNGGCWGCAG','GACTACHVGGGTATCTAATCC',p) #341F 806R\n",
    "v34_p_1=cov_e1('CCTACGGGNGGCWGCAG','GACTACHVGGGTATCTAATCC',p) #341F 806R\n",
    "v34_midas_0=cov('CCTACGGGNGGCWGCAG','GACTACHVGGGTATCTAATCC',midas) #341F 806R\n",
    "v34_midas_1=cov_e1('CCTACGGGNGGCWGCAG','GACTACHVGGGTATCTAATCC',midas) #341F 806R\n",
    "\n",
    "v4_cpr_p_0=cov(\"GTGYCAGMAGBNKCGGTVA\",\"RGACTAMNVRGGTHTCTAAT\",p)\n",
    "v4_cpr_p_1=cov_e1(\"GTGYCAGMAGBNKCGGTVA\",\"RGACTAMNVRGGTHTCTAAT\",p)\n",
    "v4_cpr_midas_0=cov(\"GTGYCAGMAGBNKCGGTVA\",\"RGACTAMNVRGGTHTCTAAT\",midas)\n",
    "v4_cpr_midas_1=cov_e1(\"GTGYCAGMAGBNKCGGTVA\",\"RGACTAMNVRGGTHTCTAAT\",midas)\n",
    "\n",
    "\n",
    "v13_c2_0=cov('AGAGTTTGATCCTGGCTCAG','ATTACCGCGGCTGCTGG',c2) #27F 534R \n",
    "v13_c2_1=cov_e1('AGAGTTTGATCCTGGCTCAG','ATTACCGCGGCTGCTGG',c2) #27F 534R \n",
    "v13_mc2_0=cov('AGAGTTTGATCCTGGCTCAG','ATTACCGCGGCTGCTGG',mc2) #27F 534R \n",
    "v13_mc2_1=cov_e1('AGAGTTTGATCCTGGCTCAG','ATTACCGCGGCTGCTGG',mc2) #27F 534R \n",
    "\n",
    "v4_c2_0=cov('GTGYCAGCMGCCGCGGTAA','GGACTACNVGGGTWTCTAAT',c2) #515F (c2arada)–806R (Ac2c2rill)\n",
    "v4_c2_1=cov_e1('GTGYCAGCMGCCGCGGTAA','GGACTACNVGGGTWTCTAAT',c2) #515F (c2arada)–806R (Ac2c2rill)\n",
    "v4_mc2_0=cov('GTGYCAGCMGCCGCGGTAA','GGACTACNVGGGTWTCTAAT',mc2) #515F (c2arada)–806R (Ac2c2rill)\n",
    "v4_mc2_1=cov_e1('GTGYCAGCMGCCGCGGTAA','GGACTACNVGGGTWTCTAAT',mc2) #515F (c2arada)–806R (Ac2c2rill)\n",
    "\n",
    "v34_c2_0=cov('CCTACGGGNGGCWGCAG','GACTACHVGGGTATCTAATCC',c2) #341F 806R\n",
    "v34_c2_1=cov_e1('CCTACGGGNGGCWGCAG','GACTACHVGGGTATCTAATCC',c2) #341F 806R\n",
    "v34_mc2_0=cov('CCTACGGGNGGCWGCAG','GACTACHVGGGTATCTAATCC',mc2) #341F 806R\n",
    "v34_mc2_1=cov_e1('CCTACGGGNGGCWGCAG','GACTACHVGGGTATCTAATCC',mc2) #341F 806R\n",
    "\n",
    "v4_cc2r_c2_0=cov(\"GTGYCAGMAGBNKCGGTVA\",\"RGACTAMNVRGGTHTCTAAT\",c2)\n",
    "v4_cc2r_c2_1=cov_e1(\"GTGYCAGMAGBNKCGGTVA\",\"RGACTAMNVRGGTHTCTAAT\",c2)\n",
    "v4_cc2r_mc2_0=cov(\"GTGYCAGMAGBNKCGGTVA\",\"RGACTAMNVRGGTHTCTAAT\",mc2)\n",
    "v4_cc2r_mc2_1=cov_e1(\"GTGYCAGMAGBNKCGGTVA\",\"RGACTAMNVRGGTHTCTAAT\",mc2)   "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d6b167b",
   "metadata": {},
   "source": [
    "# Write result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6736b442",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(r'All_Silva_phylum.tsv','w') as of:\n",
    "    of.write('Class\\tSequences number\\tV1-V3_0\\tV1-V3_e1\\tV3-V4_0\\tV3-V4_e1\\tV4-V5\\tV4-V5_e1\\tV4_0\\tV4_e1\\tV4_Mod\\tV4_Mod_e1\\n')\n",
    "    for i in p:\n",
    "        of.write(i+'\\t'+str(len(p[i]))+'\\t' \\\n",
    "                 +v13_p_0[i]+'\\t'+v13_p_1[i]+'\\t'\\\n",
    "                 +v34_p_0[i]+'\\t'+v34_p_1[i]+'\\t'\\\n",
    "                 +silva_phylum_0[i]+'\\t'+silva_phylum_1[i]+'\\t'\\\n",
    "                 +v4_p_0[i]+'\\t'+v4_p_1[i]+'\\t'\\\n",
    "                 +v4_cpr_p_0[i]+'\\t'+v4_cpr_p_1[i]+'\\n')\n",
    "with open(r'All_Silva_class.tsv','w') as of:\n",
    "    of.write('Class\\tSequences number\\tV1-V3_0\\tV1-V3_e1\\tV3-V4_0\\tV3-V4_e1\\tV4-V5\\tV4-V5_e1\\tV4_0\\tV4_e1\\tV4_Mod\\tV4_Mod_e1\\n')\n",
    "    for i in c2:\n",
    "        of.write(i+'\\t'+str(len(c2[i]))+'\\t' \\\n",
    "                 +v13_c2_0[i]+'\\t'+v13_c2_1[i]+'\\t'\\\n",
    "                 +v34_c2_0[i]+'\\t'+v34_c2_1[i]+'\\t'\\\n",
    "                 +silva_class_0[i]+'\\t'+silva_class_1[i]+'\\t'\\\n",
    "                 +v4_c2_0[i]+'\\t'+v4_c2_1[i]+'\\t'\\\n",
    "                 +v4_cc2r_c2_0[i]+'\\t'+v4_cc2r_c2_1[i]+'\\n')\n",
    "with open(r'All_midas_phylum.tsv','w') as of:\n",
    "    of.write('Class\\tSequences number\\tV1-V3_0\\tV1-V3_e1\\tV3-V4_0\\tV3-V4_e1\\tV4-V5\\tV4-V5_e1\\tV4_0\\tV4_e1\\tV4_Mod\\tV4_Mod_e1\\n')\n",
    "    for i in midas:\n",
    "        of.write(i+'\\t'+str(len(midas[i]))+'\\t' \\\n",
    "                 +v13_midas_0[i]+'\\t'+v13_midas_1[i]+'\\t'\\\n",
    "                 +v34_midas_0[i]+'\\t'+v34_midas_1[i]+'\\t'\\\n",
    "                 +midas_phylum_0[i]+'\\t'+midas_phylum_1[i]+'\\t'\\\n",
    "                 +v4_midas_0[i]+'\\t'+v4_midas_1[i]+'\\t'\\\n",
    "                 +v4_cpr_midas_0[i]+'\\t'+v4_cpr_midas_1[i]+'\\n')\n",
    "with open(r'All_midas_class.tsv','w') as of:\n",
    "    of.write('Class\\tSequences number\\tV1-V3_0\\tV1-V3_e1\\tV3-V4_0\\tV3-V4_e1\\tV4-V5\\tV4-V5_e1\\tV4_0\\tV4_e1\\tV4_Mod\\tV4_Mod_e1\\n')\n",
    "    for i in mc2:\n",
    "        of.write(i+'\\t'+str(len(mc2[i]))+'\\t' \\\n",
    "                 +v13_mc2_0[i]+'\\t'+v13_mc2_1[i]+'\\t'\\\n",
    "                 +v34_mc2_0[i]+'\\t'+v34_mc2_1[i]+'\\t'\\\n",
    "                 +midas_class_0[i]+'\\t'+midas_class_1[i]+'\\t'\\\n",
    "                 +v4_mc2_0[i]+'\\t'+v4_mc2_1[i]+'\\t'\\\n",
    "                 +v4_cc2r_mc2_0[i]+'\\t'+v4_cc2r_mc2_1[i]+'\\n')\n",
    "    "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
